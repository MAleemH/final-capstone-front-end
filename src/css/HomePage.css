* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: 'Montserrat', sans-serif;
  letter-spacing: 0.2px;
}

.list_style_none {
  list-style: none;
}

.td_none {
  text-decoration: none;
}

.home_page_main {
  display: flex;
  justify-content: space-between;
  align-items: start;
  background-color: #fffcf6;
}

.home_page_nav {
  width: 20%;
  margin: auto;
  height: 100vh;
  position: sticky;
  top: 0;
  bottom: 0;
}

.home_page_content {
  padding: 15px 0;
  width: 80%;
  margin: auto;
}

.home_page_content header {
  padding: 10px 0 20px 0;
  text-align: center;
  display: flex;
  flex-direction: column;
  gap: 20px;
}

.home_grids {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  gap: 15px;
  padding: 5px 15px;
}

.home_grids_box {
  width: 100%;
  margin: auto;
  transition: 1s;
}

.home_grids_box:hover {
  transform: scale(1.05);
}

.home_grids_box a {
  color: #172b4d;
}

.home_grids_box article {
  display: flex;
  flex-direction: column;
  gap: 7.5px;
  text-align: center;
  padding: 10px;
  background-color: #f7f0e6;
}

.therapy_figure {
  width: 100%;
  margin: auto;
  border-radius: 10px;
}

.therapy_figure img {
  width: 125px;
  object-fit: cover;
  object-position: center;
}

.home_page_socials {
  display: flex;
  justify-content: center;
  gap: 15px;
}

.home_page_socials img {
  width: 20px;
  object-fit: cover;
  object-position: center;
}
