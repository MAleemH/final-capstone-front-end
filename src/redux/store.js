import { configureStore } from '@reduxjs/toolkit';
import therapyReducer from './Therapy/therapySlice';
import userReducer from './User/userSlice';

const store = configureStore({
  reducer: {
    therapy: therapyReducer,
    user: userReducer,
  },
});

export default store;
