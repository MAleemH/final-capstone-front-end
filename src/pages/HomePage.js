import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchTherapists } from '../redux/Therapy/therapySlice';
import Navigation from '../components/Navigation';
import specializationArr from '../components/specialization';
import '../css/HomePage.css';
import Therapists from '../components/Therapists';

function HomePage() {
  const myTherapists = useSelector((state) => state.therapy.therapists);
  console.log(myTherapists);
  const dispatch = useDispatch();
  const [specialty, setSpecialty] = useState('');
  console.log(myTherapists, specialty);
  console.log(myTherapists);

  useEffect(() => {
    let active = true;
    (async () => {
      if (active && myTherapists.length === 0) {
        dispatch(fetchTherapists());
      }
    })();
    return () => {
      active = false;
    };
  }, [dispatch, myTherapists]);

  return (
    <main className="home_page_main">
      <section className="home_page_nav">
        <Navigation />
      </section>

      <section className="home_page_content">
        <header>
          <h1>Starred Therapists</h1>

          <p className="hide1">Please select a therapist</p>

          <div className="search_box">
            <fieldset className="fieldset_border_none">
              <select aria-label="Input Label" className="input_name" onChange={(e) => setSpecialty(e.target.value)} id="specializationId">

                {specializationArr.map((specialty) => (
                  <option key={specialty} value={specialty.value} aria-label="Input Specialization">{specialty.name}</option>
                ))}
              </select>
            </fieldset>
          </div>
        </header>

        <div>
          <Therapists myTherapists={myTherapists} />
        </div>

      </section>

    </main>
  );
}

export default HomePage;
