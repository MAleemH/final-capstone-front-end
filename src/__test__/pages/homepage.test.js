import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../../redux/store';
import HomePage from '../../pages/HomePage';

it('Home page renders correctly', () => {
  const myTherapists = store.getState().therapy.therapists;
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <HomePage myTherapists={myTherapists} />
      </MemoryRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
