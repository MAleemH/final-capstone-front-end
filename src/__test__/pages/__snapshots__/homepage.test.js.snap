// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page renders correctly 1`] = `
<main
  className="home_page_main"
>
  <section
    className="home_page_nav"
  >
    <main>
      <section
        className="home_nav_section"
      >
        <header
          className="home_logo"
        >
          <figure>
            <a
              className="td_none"
              href="/homepage"
              onClick={[Function]}
            >
              <img
                alt=""
                src="logo.png"
              />
            </a>
          </figure>
        </header>
        <nav
          className="list_style_none home_nav"
        >
          <li>
             
            <a
              activeClassName="active"
              className="td_none"
              href="/homepage"
              onClick={[Function]}
            >
              HOME
            </a>
          </li>
          <li>
             
            <a
              activeClassName="active"
              className="td_none"
              href="/appointments"
              onClick={[Function]}
            >
              APPOINTMENTS
            </a>
          </li>
          <li>
             
            <a
              activeClassName="active"
              className="td_none"
              href="/book"
              onClick={[Function]}
            >
              NEW APPOINTMENT
            </a>
          </li>
          <li>
            <button
              className="logout_btn"
              onClick={[Function]}
              type="button"
            >
              LOGOUT
            </button>
          </li>
        </nav>
        <footer
          className="home_nav_footer"
        >
          <figure
            className="nav_social_icons"
          >
            <a
              className="td_none"
              href="/"
              onClick={[Function]}
            >
              <img
                alt=""
                src="facebook.png"
              />
            </a>
          </figure>
          <figure
            className="nav_social_icons"
          >
            <a
              className="td_none"
              href="/"
              onClick={[Function]}
            >
              <img
                alt=""
                src="twitter.png"
              />
            </a>
          </figure>
          <figure
            className="nav_social_icons"
          >
            <a
              className="td_none"
              href="/"
              onClick={[Function]}
            >
              <img
                alt=""
                src="linkedin.png"
              />
            </a>
          </figure>
          <figure
            className="nav_social_icons"
          >
            <a
              className="td_none"
              href="/"
              onClick={[Function]}
            >
              <img
                alt=""
                src="instagram.png"
              />
            </a>
          </figure>
          <figure
            className="nav_social_icons"
          >
            <a
              className="td_none"
              href="/"
              onClick={[Function]}
            >
              <img
                alt=""
                src="snapchat.png"
              />
            </a>
          </figure>
        </footer>
      </section>
      <section
        className="home_nav_section_mobile"
      >
        <header
          className="home_logo margin_left1"
        >
          <button
            aria-label="button"
            onClick={[Function]}
            type="button"
          >
            <img
              alt=""
              src="menu.png"
            />
          </button>
        </header>
      </section>
    </main>
  </section>
  <section
    className="home_page_content"
  >
    <header>
      <h1>
        Starred Therapists
      </h1>
      <p
        className="hide1"
      >
        Please select a therapist
      </p>
      <div
        className="search_box"
      >
        <fieldset
          className="fieldset_border_none"
        >
          <select
            aria-label="Input Label"
            className="input_name"
            id="specializationId"
            onChange={[Function]}
          >
            <option
              aria-label="Input Specialization"
              value=""
            >
              Select Specialization
            </option>
            <option
              aria-label="Input Specialization"
              value="Cardiovascular"
            >
              Cardiovascular
            </option>
            <option
              aria-label="Input Specialization"
              value="Emotional"
            >
              Emotional
            </option>
            <option
              aria-label="Input Specialization"
              value="Romance"
            >
              Romance
            </option>
            <option
              aria-label="Input Specialization"
              value="Sexual"
            >
              Sexual
            </option>
            <option
              aria-label="Input Specialization"
              value="Academic"
            >
              Academic
            </option>
            <option
              aria-label="Input Specialization"
              value="Career"
            >
              Career
            </option>
            <option
              aria-label="Input Specialization"
              value="Marital"
            >
              Marital
            </option>
            <option
              aria-label="Input Specialization"
              value="Dating"
            >
              Dating
            </option>
          </select>
        </fieldset>
      </div>
    </header>
    <div>
      <div
        className="home_grids"
      />
    </div>
  </section>
</main>
`;
